modules/ssh/manifests/config.pp

```diff
@@ -0,0 +1,117 @@
+                'RSAkey':
+                    command => '"C:\Program Files\KTS\daemon.exe" -rsakey',
+                    require => Class['packages::kts'],
+                    notify  => Service['KTS'],
+                    creates => $rsakey_ky;
+            }
+
+            # install the logon batch file which will share the MOTD file
+            file {
+                'C:/Program Files/KTS/Scripts/allusers.bat':
+                    source  => 'puppet:///modules/ssh/allusers.bat',
+                    replace => true,
+                    require => Class['packages::kts'];
+            }
+
+            # regenerate publickey_logon_ini; this is connected to everything
+            # that would modify C:\Program Files\KTS\*.keys (from
+            # ssh::userconfig and ssh::extra_authorized_key) or C:\Program
+            # Files\KTS\*.pass (from users::{root,builder,etc.}
+            file {
+                "${::ssh::settings::genkey_dir}/genkeys.rb":
+                    source  => 'puppet:///modules/ssh/genkeys.rb',
+                    require => Class['packages::kts'];
+            }
+            exec {
+                'generate-kts-publickey-logon-ini':
+                    command     => "\"${::ruby_interpreter}\" genkeys.rb -o ../publickey_logon.ini",
+                    cwd         => $ssh::settings::genkey_dir,
+                    require     => File["${::ssh::settings::genkey_dir}/genkeys.rb"],
+                    refreshonly => true,
+                    logoutput   => true;
+            }
+
+            # finally, ensure that all of the files with passwords in them are
+            # readable only by SYSTEM and root (and in particular, not the builder)
+            acl {
+                [$publickey_logon_ini, $ssh::settings::genkey_dir, $rsakey_ky, $kts_ini]:
+                    purge                      => true,
+                    inherit_parent_permissions => false,
+                    require                    => [ Exec['generate-kts-publickey-logon-ini'],
+                                                    File[$ssh::settings::genkey_dir],
+                                                    Exec[ 'RSAkey'],
+                                                    File[$kts_ini],
+                                                  ],
+                    # indentation of => is not properly aligned in hash within array: The solution was provided on comment 2
+                    # https://github.com/rodjek/puppet-lint/issues/333
+                    permissions                => [
+                        {
+                          identity => 'root',
+                          rights   => ['full'] },
+                        {
+                          identity => 'SYSTEM',
+                          rights   => ['full'] },
+                    ];
+            }
+        }
+
+        default: {
+            # Duo MFA requires special sshd configuration
+            $sshd_config_content = $duo_enabled ? {
```

Comment:
We aren't utilizing duo in this repo and when we do, we won't be using node scope vars.  So remove this.